core/src/main/scala/kafka/common/AuthorizationException.scala

```diff
@@ -16,21 +16,27 @@
  */
 package kafka.common
 
+import org.apache.kafka.common.errors
+
 /**
  * Exception thrown when a principal is not authorized to perform an operation.
  * @param message
  */
-abstract class AuthorizationException(message: String) extends RuntimeException(message) {
+@deprecated("Replaced by org.apache.kafka.common.errors.AuthorizationException", "12-09-2015")
+abstract class AuthorizationException(message: String) extends errors.AuthorizationException(message) {
 }
 
-class TopicAuthorizationException(message: String) extends AuthorizationException(message) {
+@deprecated("Replaced by org.apache.kafka.common.errors.TopicAuthorizationException", "12-09-2015")
```

Comment:
Aren't changes to the inheritance hierarchy like this not backwards compatible? We've effectively removed `kafka.common.AuthorizationException` from the hierarchy of `kafka.common.TopicAuthorization` and friends. It's only ok to add new classes into the hierarchy. I'm not sure you'd actually see errors when running, but at a minimum some `try`/`catch` clauses which should be valid would no longer work.

Given that Java makes it hard to rectify two separate class hierarchies, I'd probably suggest avoiding trying to unify the two at all -- just move as many non-public, server-internal usages to the new versions and leave the rest to be cleaned up along with the deprecation and removal of the old client classes.